src/DssSpell.t.base.sol

```diff
@@ -1826,48 +1814,175 @@ contract DssSpellTestBase is Config, DssTest {
-        address[] memory _chainlog_addrs = new address[](_count);
+        string memory pversion = chainLog.version();
+        bytes32[] memory pkeys = chainLog.list();
+        uint256 pcount = pkeys.length;
+        address[] memory pvalues = new address[](pcount);
 
-        for(uint256 i = 0; i < _count; i++) {
-            (, address _val) = chainLog.get(i);
-            _chainlog_addrs[i] = _val;
+        for(uint256 i = 0; i < pcount; i++) {
+            pvalues[i] = chainLog.getAddress(pkeys[i]);
         }
 
         _vote(address(spell));
         _scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        if (keccak256(abi.encodePacked(_version)) == keccak256(abi.encodePacked(chainLog.version()))) {
+        if (keccak256(abi.encodePacked(pversion)) == keccak256(abi.encodePacked(chainLog.version()))) {
+            bytes32[] memory keys = chainLog.list();
+            uint256 count = keys.length;
+            address[] memory values = new address[](count);
+
             // Fail if the version is not updated and the chainlog count has changed
-            if (_count != chainLog.count()) {
-                emit log_named_string("Error", _concat("TestError/chainlog-version-not-updated-count-change-", _version));
-                fail();
-                return;
-            }
-            // Fail if the chainlog is the same size but local keys don't match the chainlog.
-            for(uint256 i = 0; i < _count; i++) {
-                (, address _val) = chainLog.get(i);
-                if (_chainlog_addrs[i] != _val) {
-                    emit log_named_string("Error", _concat("TestError/chainlog-version-not-updated-address-change-", _version));
-                    fail();
-                    return;
-                }
+            assertEq(count, pcount, "TestError/chainlog-version-not-updated-count-change");
+
+            for(uint256 i = 0; i < count; i++) {
+                values[i] = chainLog.getAddress(keys[i]);
+                // Fail if the chainlog is the same size, but any address or their order changed.
+                assertEq(
+                    values[i],
+                    pvalues[i],
+                    _concat("TestError/chainlog-version-not-updated-address-change: ", keys[i])
+                );
             }
         }
     }
 
+    function _testNewOrUpdatedChainlogValues(string[] memory keys) internal {
+        // `setAddress` must have being called `keys.length` times.
+        vm.expectCall(
+            address(chainLog),
+            abi.encodeWithSelector(chainLog.setAddress.selector),
+            uint64(keys.length)
+        );
+
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        for (uint256 i = 0; i < keys.length; i++) {
+            _checkAuth(_stringToBytes32(keys[i]));
+        }
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[1] memory keys) internal {
```

Comment:
This is a limitation of Solidity. One cannot initialize a dynamic sized array with a literal array declaration:

```solidity
bytes32[] memory keys = [
  "KEY_A",
  "KEY_B",
];
```

Having these overloads allows us to call the function with the literal notation:
```
        _testNewOrUpdatedChainlogValues([
            "KEY_A",
            "KEY_B"
        ]);
```

Otherwise, we'd have to manually change the length of the array and assign each item individually:

```
        bytes32[] memory keys = new bytes32[](2);
        keys[0] = "KEY_A";
        keys[1] = "KEY_B";
        _testNewOrUpdatedChainlogValues(keys);
```

Honestly, I thought this would be a DX improvement, but I don't feel that strong about it.

Reply:
Yeah, I personally vote for manually changing the length of the array (as we do in multiple places in the codebase) instead of redeclaring the function multiple times